### PR TITLE
feat(cmd) ask confirmation from user before overwriting a file

### LINKED ISSUE #184: Deck should not overwrite existing state file without asking
Currently, `deck dump` will overwrite an existing state file without asking - I have lost some configuration that way.

E.g.:
```ShellCommand
$ deck dump
$ ls -hl
total 248
-rw-------  1 cwurm  wheel   121K  7 Jul 10:43 kong.yaml

$ deck dump -w Shared
$ ls -hl
total 8
-rw-------  1 cwurm  wheel   2.9K  7 Jul 10:44 kong.yaml
```

I think that ideally, `deck dump` would ask before overwriting an existing file, similar to how `deck reset` asks, and with a `-f` parameter to force it.

### PR DESCRIPTION
**Pull request by hbagdi**
Fix #184

### CODE REVIEW

File: cmd/dump.go [hunk 1]
```diff
@@ -107,6 +107,14 @@ configure Kong.`,
 			return nil
 		}
 
+		ok, err := confirmFileOverwrite(dumpCmdKongStateFile, dumpCmdStateFormat, assumeYes)
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return nil
+		}
```
Comment (mflendrich) [inline cmd/dump.go:116]:
```suggestion
		if ok, err := confirmFileOverwrite(dumpCmdKongStateFile, dumpCmdStateFormat, assumeYes); err != nil {
			return err
		} else if !ok {
			return nil
		}
```

nit - the variables don't need to be function-scoped

Also another nit: `ok` is a name in Go that has a well-defined meaning (meaning that a type assertion or key lookup succeeded). How about changing this to `hasConsent`?

Also applies to other repetitions.

**Review by mflendrich — APPROVED** [review]
Some non-blocking bikeshedding attached. LGTM, apply optionally

**Comment by hbagdi** [comment]
@mflendrich I had to resolve a couple of conflicts too, mind doing another pass?

File: cmd/utils.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	"os"
+	"strings"
+
+	"github.com/kong/deck/utils"
+)
+
+// confirm prompts a user for a confirmation with message
+// and returns true with no error if input is "yes" or "y" (case-insensitive),
+// otherwise false.
+func confirm(message string) (bool, error) {
```
Comment (mflendrich) [inline cmd/utils.go:15]: Have you considered using a library?

For example:

https://pkg.go.dev/gopkg.in/AlecAivazis/survey.v1#Confirm

File: cmd/konnect_dump.go [hunk 3]
```diff
@@ -31,6 +31,14 @@ configure Konnect.` + konnectAlphaState,
...
 		httpClient := http.DefaultClient
 
+		ok, err := confirmFileOverwrite(dumpCmdKongStateFile, dumpCmdStateFormat, assumeYes)
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return nil
+		}
+
```
Comment (mflendrich) [inline cmd/konnect_dump.go:41]: Could this block be made identical to the equivalent one in dump.go?

File: cmd/utils.go (hunk 2 above)
Comment (hbagdi) [reply cmd/utils.go:15]: We don't do much here so I avoided adding any dependencies.

File: cmd/konnect_dump.go (hunk 3 above)
Comment (hbagdi) [reply cmd/konnect_dump.go]: Updated.

**Review by mflendrich — APPROVED** [review]

**Pull request merged**